iroh-unixfs/src/builder.rs

```diff
@@ -522,15 +522,16 @@ impl DirectoryBuilder {
 
         ensure!(typ == DirectoryType::Basic, "too many links to fit into one chunk, must be encoded as a HAMT. However, HAMT creation has not yet been implemented.");
 
-        let name = name.unwrap_or_default();
-
-        if let Some(path) = path {
+        Ok(if let Some(path) = path {
             let mut dir = make_dir_from_path(path, chunker.clone(), degree).await?;
-            dir.name = name;
-            return Ok(dir);
-        }
-
-        Ok(Directory { name, entries })
+            if let Some(name) = name {
+                dir.name = name;
+            }
+            dir
+        } else {
+            let name = name.unwrap_or_default();
```

Comment:
This fixes the issue, but nevertheless this is all a bit weird. Feels like this should be two different builders, one that builds from a path and one that builds from its contents.

```
enum Builder {
  Path { ... }
  Parts { ... }
}
```

Otherwise, e.g. if entries are set and path is set, the entries are being ignored.